drivers/modem/gsm_ppp.c

```diff
@@ -1168,6 +1150,10 @@ static int gsm_init(const struct device *dev)
 			   K_PRIO_COOP(7), NULL);
 	k_thread_name_set(&gsm->workq.thread, "gsm_workq");
 
+#if defined(CONFIG_GSM_MUX)
```

Comment:
Why not `if (IS_ENABLED(CONFIG_GSM_MUX)`?

Reply:
I noticed that the `rssi_work_handle` is declared in the struct but is unused if `CONFIG_GSM_MUX=n`, if we exclude that then we have to leave the macro as is, otherwise we can convert quite a few `#if defined(CONFIG_GSM_MUX)` to `if (IS_ENABLED(CONFIG_GSM_MUX)`

Reply:
I decided to keep this as-is for now. We will decide what to do with them in the next PR